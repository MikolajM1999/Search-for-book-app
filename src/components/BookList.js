import React from 'react'
import BookListItem from './BookListItem'

const BookList = (props) => (
   <div>
      {props.searchingBooks.length === 0 ? (
         <div>
            <span>Search for your book, fill at least title field.</span>
         </div>
      ) : (
            props.searchingBooks.map((book) => {
               return <BookListItem key={book.bookId} {...book} />
            })
         )}
   </div>
)

export default BookList