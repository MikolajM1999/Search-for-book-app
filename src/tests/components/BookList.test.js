import React from 'react'
import { shallow } from 'enzyme'
import BookList from '../../components/BookList'
import books from '../fixtures/books'

test('BookList renders without crashing with default array value', () => {
   shallow(<BookList searchingBooks={[]} />)
})

test('BookList renders without crashing with books', () => {
   shallow(<BookList searchingBooks={books} />)
})

test('should render BookList correctly with books', () => {
   const wrapper = shallow(<BookList searchingBooks={books} />)
   expect(wrapper).toMatchSnapshot()
})

test('should render BookList correctly with default empty array', () => {
   const wrapper = shallow(<BookList searchingBooks={[]} />)
   expect(wrapper).toMatchSnapshot()
})