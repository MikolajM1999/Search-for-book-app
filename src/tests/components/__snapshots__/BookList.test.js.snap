// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BookList correctly with books 1`] = `
<div
  className="container-for-books"
>
  <BookListItem
    bookCanonicalVolumeLink="https://www.google.com/"
    bookFullTitle="Title for testing 1"
    bookId="1"
    bookImageLink="./img/unknown_book.png"
    bookShortDescription="word1 word2 word3 word4 word5 word6 word7 word8 word9 word10 word11 word12 word13 word14 word15 word16 word17 word18 word19 word20 word21."
    key="1"
    publishedDate="1971"
  />
  <BookListItem
    bookCanonicalVolumeLink="https://www.google.com/"
    bookFullTitle="Title for testing 2"
    bookId="2"
    bookImageLink="./img/unknown_book.png"
    bookShortDescription="word1 word2 word3 word4 word5 word6 word7 word8 word9 word10 word11 word12 word13 word14 word15 word16 word17 word18 word19 word20 word21 word22."
    key="2"
    publishedDate="1972"
  />
  <BookListItem
    bookCanonicalVolumeLink="https://www.google.com/"
    bookFullTitle="Title for testing 3"
    bookId="3"
    bookImageLink="./img/unknown_book.png"
    bookShortDescription="word1 word2 word3 word4 word5 word6 word7 word8 word9 word10 word11 word12 word13 word14 word15 word16 word17 word18 word19 word20 word21 word22 word23."
    key="3"
    publishedDate="1973"
  />
</div>
`;

exports[`should render BookList correctly with default empty array 1`] = `
<div
  className="container-for-books"
>
  <div
    className="info"
  >
    <span
      className="info__message"
    >
      Search for your book, fill at least title field
    </span>
  </div>
</div>
`;
